projects/storefrontlib/src/cms-components/navigation/page-header/page-header.component.ts

```diff
@@ -0,0 +1,34 @@
+import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import {
+  CmsPageHeaderComponent,
+  PageMeta,
+  PageMetaService,
+} from '@spartacus/core';
+import { Observable } from 'rxjs';
+import { filter, map } from 'rxjs/operators';
+import { CmsComponentData } from '../../../cms-structure/page/model/cms-component-data';
+
+@Component({
+  selector: 'cx-page-header',
+  templateUrl: './page-header.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class PageHeaderComponent implements OnInit {
+  title$: Observable<string>;
+
+  constructor(
+    public component: CmsComponentData<CmsPageHeaderComponent>,
+    protected pageMetaService: PageMetaService
+  ) {}
+
+  ngOnInit(): void {
+    this.setTitle();
+  }
```

Comment:
I think there is a convention of moving away from using ngOnInit() for this sort off thing.  If we do move away from ngOnInit, we'd have to make sure it's not a breaking change.

Reply:
@plabadie paradoxically, kicking it from ngOnInit will be breaking changes😅 Please check BreadcrumbComponent changes. I removed this stuff from there, and add inheritance. It means that now its safe for current implementations

Reply:
Agreed, let's play it safe and leave it as-is.